### PR TITLE
fix deprecated analyzer

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

**Comment by kevmoo** [comment]
Fixed, @natebosch

File: json_serializable/lib/src/json_key_utils.dart [hunk 1]
```diff
@@ -68,7 +68,7 @@ JsonKey _from(FieldElement element, JsonSerializable classAnnotation) {
       // TODO(kevmoo): Support calling function for the default value?
       badType = 'Function';
     } else if (!reader.isLiteral) {
-      badType = dartObject.type.name;
+      badType = dartObject.type.getDisplayString();
```
Comment (jakemac53) [inline json_serializable/lib/src/json_key_utils.dart]: Lets be consistent about `.element.name`.

Comment (kevmoo) [reply json_serializable/lib/src/json_key_utils.dart]: fixed

Comment (jakemac53) [reply json_serializable/lib/src/json_key_utils.dart]: The others too :)

**Comment by kevmoo** [comment]
> hmm, previously this was a cascade which should have had no effect? This might have actually changed the behavior.

The cascades were copy-paste mistakes. Look at the diff now. Much better!

**Pull request merged**